File: WLLM/wwwroot/Notificaciones_Mensajeria/Model/NotificationRequest.js
//@ts-check

import { ModelFiles } from "../../WDevCore/WModules/CommonModel.js";
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
class NotificationRequest extends EntityClass {
    /** @param {Partial<NotificationRequest>} [props] */
    constructor(props) {
        super(props, 'Notificaciones');
        for (const prop in props) {
            this[prop] = props[prop];
        };
    }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Titulo;
   /**@type {String}*/ Mensaje;
   /**@type {Array<ModelFiles>}*/ Files;
   /**@type {String}*/ NotificationType;
   /**@type {Array<Number>}*/ Usuarios;
   /**@type {Array<Number>}*/ Dependencias;
   /**@type {Array<Number>}*/ ToAdress; //correos o numeros de telefono
   /**@type {String}*/ NotificationsServicesEnum;  
   /**@type {Array<NotificactionDestinatarios>}*/  Destinatarios;
}

export { NotificationRequest }

export class NotificactionDestinatarios {
    /**@type {String} */ Correo;
    /**@type {String} */ Telefono;
    /**@type {Object} */ NotificationData;
}

File: BusinessLogic/Notificaciones_Mensajeria/Gestion_Notificaciones/Model/NotificationRequest.cs
using API.Controllers;
using APPCORE;
using APPCORE.Security;
using APPCORE.Services;
using CAPA_NEGOCIO.Gestion_Mensajes.Operations;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel
{
	public class NotificationRequest
	{
		public string? MediaUrl { get; set; }//creo que no se va utilizar
		public string? Titulo { get; set; }
		public string? Mensaje { get; set; }
		public List<ModelFiles>? Files { get; set; }
		public NotificationTypeEnum? NotificationType { get; set; }
		public bool? EsResponsable { get; set; }
		public List<NotificactionDestinatarios>? Destinatarios { get; set; }
		public List<int?>? Usuarios { get; set; }
		public List<int?>? Dependencias { get; set; }
		public List<string>? ToAdress { get; set; }//para mensajes directos sin usuarios correos o numeros de telefono
		public List<NotificationsServicesEnum>? NotificationsServices { get; set; }
	}
	public enum NotificationsServicesEnum
	{
		WHATSAPP, MAIL
	} 

	public enum NotificationTypeEnum
	{
		USUARIOS, DEPENDENCIA, LIBRE
	}
}